metricbeat/module/zookeeper/mntr/data.go

```diff
@@ -0,0 +1,155 @@
+package mntr
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strconv"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func eventMapping(response io.Reader) common.MapStr {
+
+	var (
+		versionString           string
+		serverState             string
+		avgLatency              int
+		minLatency              int
+		maxLatency              int
+		packetsReceived         int
+		packetsSent             int
+		numAliveConnections     int
+		outstandingRequests     int
+		znodeCount              int
+		watchCount              int
+		ephemeralsCount         int
+		approximateDataSize     int
+		openFileDescriptorCount int
+		maxFileDescriptorCount  int
+		followers               int
+		syncedFollowers         int
+		pendingSyncs            int
+	)
+
+	//  zk_version      3.5.1-alpha-1693007, built on 07/28/2015 07:19 GMT
+	//  zk_avg_latency  0
+	//  zk_max_latency  1789
+	//  zk_min_latency  0
+	//  zk_packets_received     22152032
+	//  zk_packets_sent 30959914
+	//  zk_num_alive_connections        1033
+	//  zk_outstanding_requests 0
+	//  zk_server_state leader
+	//  zk_znode_count  242609
+	//  zk_watch_count  940522
+	//  zk_ephemerals_count     8565
+	//  zk_approximate_data_size        372143564
+	//  zk_open_file_descriptor_count   1083
+	//  zk_max_file_descriptor_count    1048576
+	//  zk_followers    5
+	//  zk_synced_followers     2
+	//  zk_pending_syncs        0
+
+	var re *regexp.Regexp
+	scanner := bufio.NewScanner(response)
+
+	for scanner.Scan() {
+
+		re = regexp.MustCompile("zk_version\\s+(.*$)")
```

Comment:
It would be better to do `regexp.MustCompile` one time in the `init()` method. This guarantees that the regular expressions compile when the app starts and not the first time an event is received. It also prevents new `Regexp` objects from being created each time an event is processed. Regexp objects are safe for concurrent use so that can be stored statically.